uint8 DIRECTION_FORWARDS = 0
uint8 DIRECTION_BACKWARDS = 1
uint8 direction
---
